src/uct/ib/ud/base/ud_iface.h

```diff
@@ -410,7 +401,25 @@ uct_ud_iface_check_grh(uct_ud_iface_t *iface, void *packet, int is_grh_present)
         return 1;
     }
 
-    gid_len = uct_ud_grh_get_dgid_len(grh);
+    /*
+     * Take the packet type from CQE, because:
+     * 1. According to Annex17_RoCEv2 (A17.4.5.1):
+     * For UD, the Completion Queue Entry (CQE) includes remote address
+     * information (InfiniBand Specification Vol. 1 Rev 1.2.1 Section 11.4.2.1).
+     * For RoCEv2, the remote address information comprises the source L2
+     * Address and a flag that indicates if the received frame is an IPv4,
+     * IPv6 or RoCE packet.
+     * 2. According to PRM, for responder UD/DC over RoCE sl represents RoCE
+     * packet type as:
+     * bit 3    : when set R-RoCE frame contains an UDP header otherwise not
+     * Bits[2:0]: L3_Header_Type, as defined below
+     *     - 0x0 : GRH - (RoCE v1.0)
+     *     - 0x1 : IPv6 - (RoCE v1.5/v2.0)
+     *     - 0x2 : IPv4 - (RoCE v1.5/v2.0)
+     */
+    gid_len = ((roce_pkt_type & 0x7) == 0x2) ? UCS_IPV4_ADDR_LEN :
```

Comment:
maybe UCT_IB_MLX5_CQE_SL_PKTYPE_MASK should be 0xc000..0 ?

Reply:
in ud verbs I get the whole `wc->sl` which is pkt_type in case of RoCE.
So I have to mask it here anyway

Reply:
so mask the sl before passing it to here

Reply:
fixed, used 0x7, because PRM defines 3 bits for HDR type